test/Sign.SignatureProviders.KeyVault.Test/KeyVaultServiceTests.cs

```diff
@@ -56,9 +61,72 @@ public void Constructor_WhenCertificateNameIsEmpty_Throws()
         public void Constructor_WhenLoggerIsNull_Throws()
         {
             ArgumentNullException exception = Assert.Throws<ArgumentNullException>(
-                () => new KeyVaultService(CertificateClient, CryptographyClient, CertificateName, logger: null!));
+                () => new KeyVaultService(_certificateClient.Object, _cryptographyClient.Object, CertificateName, logger: null!));
 
             Assert.Equal("logger", exception.ParamName);
         }
+
+        [Fact]
+        public async Task GetCertificateAsync_CalledTwice_CertificateRetrievedOnce()
+        {
+            CancellationToken cancellationToken = CancellationToken.None;
+            Mock<KeyVaultCertificateWithPolicy> certificate = CreateMockKeyVaultCertificateWithPolicy();
+            Mock<Response<KeyVaultCertificateWithPolicy>> response = new();
+
+            response
+                .Setup(_ => _.Value)
+                .Returns(certificate.Object);
+
+            _certificateClient
+                .Setup(_ => _.GetCertificateAsync(CertificateName, cancellationToken))
+                .ReturnsAsync(response.Object);
+
+            using KeyVaultService service = new(_certificateClient.Object, _cryptographyClient.Object, CertificateName, Logger);
+
+            using X509Certificate2 certificate1 = await service.GetCertificateAsync(cancellationToken);
+            using X509Certificate2 certificate2 = await service.GetCertificateAsync(cancellationToken);
+
+            _certificateClient.Verify(_ => _.GetCertificateAsync(CertificateName, cancellationToken), Times.Once);
+        }
+
+        [Fact]
+        public async Task GetRsaAsync_ReturnsRSAKeyVaultWrapper()
+        {
+            CancellationToken cancellationToken = CancellationToken.None;
+            Mock<KeyVaultCertificateWithPolicy> certificate = CreateMockKeyVaultCertificateWithPolicy();
+            Mock<RSAKeyVault> rsaKeyVault = new(Mock.Of<CryptographyClient>(), "testId", null);
+            Mock<Response<KeyVaultCertificateWithPolicy>> response = new();
+
+            response
+                .Setup(_ => _.Value)
+                .Returns(certificate.Object);
+
+            _certificateClient
+                .Setup(_ => _.GetCertificateAsync(CertificateName, cancellationToken))
+                .ReturnsAsync(response.Object);
+
+            _cryptographyClient
+                .Setup(_ => _.CreateRSAAsync(cancellationToken))
+                .ReturnsAsync(rsaKeyVault.Object);
+
+            using KeyVaultService service = new(_certificateClient.Object, _cryptographyClient.Object, CertificateName, Logger);
+
+            using RSA rsa = await service.GetRsaAsync(cancellationToken);
+        }
```

Comment:
Can you assert that `rsa` is an instance of `RSAKeyVaultWrapper`?

Reply:
I just pushed a patch that adds this assert.